src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -1235,6 +1244,7 @@ private function createDOMXPath(\DOMDocument $document, array $prefixes = []): \
             $namespace = $this->discoverNamespace($domxpath, $prefix);
             if (null !== $namespace) {
                 $domxpath->registerNamespace($prefix, $namespace);
+                $this->cachedNamespaces[$prefix] = $namespace;
```

Comment:
I would do this caching in `discoverNamespace` instead, so that the caching management is centralized in the place doing memoization (and it could cache `null` values too)

Reply:
Good point 👍